### PR TITLE
Updating JnlpAccessWithSecuredHudsonTest to JenkinsRule

### PR DESCRIPTION
**Pull request by jglick**
No changelog, etc., just internal code cleanup.

@reviewbybees

### CODE REVIEW

File: test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java [hunk 1]
```diff
@@ -92,47 +106,43 @@ public void testAnonymousCanAlwaysLoadJARs() throws Exception {
...
+        FileUtils.copyURLToFile(new Slave.JnlpJar("slave.jar").getURL(), slaveJar);
+        Proc p = new hudson.Launcher.LocalLauncher(StreamTaskListener.fromStderr()).launch().
+            stdout(System.out).stderr(System.err).
+            cmds(JavaEnvUtils.getJreExecutable("java"), "-jar", slaveJar.getAbsolutePath(), "-jnlpUrl", r.getURL() + "computer/test/slave-agent.jnlp", "-secret", secret).
+            start();
         try {
-            for (int i = 0; i < /* one minute */600; i++) {
-                if (slave.getComputer().isOnline()) {
-                    System.err.println("JNLP slave successfully connected");
-                    Channel channel = slave.getComputer().getChannel();
```
Comment (jglick) [inline test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java]: [diff for review](https://github.com/jenkinsci/jenkins/pull/3096/files?w=1#diff-139a67312141a030a142c3f3f69da215L115)

File: test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java [hunk 2]
```diff
@@ -92,47 +106,43 @@ public void testAnonymousCanAlwaysLoadJARs() throws Exception {
...
-                    assertTrue(f.exists());
-                    try {
-                        fail("SECURITY-206: " + channel.call(new Attack(f.getAbsolutePath())));
-                    } catch (SecurityException x) {
-                        System.out.println("expected: " + x);
-                        assertTrue(x.getMessage().contains("https://jenkins.io/redirect/security-144"));
-                    }
-                    return;
-                }
+            while (!slave.toComputer().isOnline()) { // TODO can use r.waitOnline(slave) after https://github.com/jenkinsci/jenkins-test-harness/pull/80
```
Comment (jglick) [inline test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java]: https://github.com/jenkinsci/jenkins-test-harness/pull/80

File: test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java [hunk 3]
```diff
@@ -69,13 +82,14 @@ protected Slave createNewJnlpSlave(String name) throws Exception {
 
     @PresetData(DataSet.NO_ANONYMOUS_READACCESS)
     @Email("http://markmail.org/message/on4wkjdaldwi2atx")
```
Comment (daniel-beck) [inline test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java]: We're using the weirdest annotations.

**Review by daniel-beck — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
🐝 and @reviewbybees done

**Pull request merged**